### PR TITLE
WEEBUG-2795 - Add 'indicated' object to highlight a specific object on the canvas

### PR DESCRIPTION
**Pull request by msucharski**
Add a way to indicate an object (currently just used for textboxes to bring hovered-box to the front to indicate which box will be selected when the user clicks). Without this it is a little hard to see the edges of a textbox if there are multiple overlapping boxes. This also brings it to the 'top' by rendering it after other objects.

### CODE REVIEW

File: dist/fabric.js [hunk 1]
```diff
@@ -17572,6 +17645,37 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
...
+        -width / 2,
+        -height / 2,
+        width,
+        height
+      );
+
+      ctx.restore();
+      return this;
+    },
+
```
Comment (cdeng-amplify) [inline dist/fabric.js:17678]: still need to keep this around?

**Review by cdeng-amplify — APPROVED** [review]
looks good, just a minor comment about commented code.

File: dist/fabric.js [hunk 2]
```diff
@@ -17637,6 +17741,22 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
...
+     * @return {fabric.Object} thisArg
+     * @chainable
+     */
+    drawIndication: function(ctx, styleOverride) {
+      styleOverride = styleOverride || {};
+      this.drawIndicationBorders(ctx, styleOverride);
+      return this;
+    },
+
+
```
Comment (cdeng-amplify) [inline dist/fabric.js:17759]: this also.

File: dist/fabric.js [hunk 3]
```diff
@@ -27251,6 +27371,13 @@ fabric.Image.filters.BaseFilter.fromObject = function(object, callback) {
      */
     editingBorderColor: 'rgba(102,153,255,0.25)',
 
+    /**
+     * Border color of text object while indicated
+     * @type String
+     * @default
+     */
+    indicationBorderColor: 'rgba(00,254,0,0.25)',
+
```
Comment (cdeng-amplify) [inline dist/fabric.js:27380]: and this.

File: dist/fabric.js (hunk 1 above)
Comment (msucharski) [reply dist/fabric.js:17678]: Yeah, this is the format for all functions and object variables - Note that it's just the documentation at the top for the description and params that is commented (not sure why github is coloring it ALL as comment....)

File: dist/fabric.js (hunk 2 above)
Comment (msucharski) [reply dist/fabric.js:17759]: Same as above (funny that the documentation is more lines than the actual code!)

File: dist/fabric.js (hunk 3 above)
Comment (msucharski) [reply dist/fabric.js:27380]: Yup! Same documentation format as the other vars in the file.

**Review by Jvogel-Amplify — APPROVED** [review]

**Review by brassaro — APPROVED** [review]
Same annoying comment as for the Classroom PR: wouldn't "focus" instead of "indicated" be a little less clunky of a word? Says the non-English native speaker...

**Comment by msucharski** [comment]
> Same annoying comment as for the Classroom PR: wouldn't "focus" instead of "indicated" be a little less clunky of a word? Says the non-English native speaker...

I'm copy-pasting this from my reply in the other PR - sorry for the redundancy but I think this conversation makes more sense in this repo so: 

I'm glad you ask! Actually I spent a lot of time trying to figure out how to name this. Fabricjs already has a 'highlight' (and a 'hovered') notion but they are both used for different reasons. 'Focused' might make sense but it also has a connotation with HTML that I would want to avoid (in fact the way the textboxes 'work' is to add focus on a hidden HTML text element so this might confuse that even more!) I realize 'indicated' is a bit vague but it was the closest non-taken description I could think of for 'indicating an object which may become selected if you take further action'

**Pull request merged**